src/components/text-input-email.component.js

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable unicorn/better-regex */
+import $ from 'jquery';
+import TextInput from './text-input.components';
+
+export default class EmailTextInput extends TextInput {
+    constructor(stringQuery, textEnable = false, required = true) {
+        super(stringQuery, textEnable, required);
+        this.emailInput = $(`${stringQuery} input[type="email"]`);
+    }
+
+    getEmailInput() {
+        return this.emailInput.val();
+    }
+
+    setEmailInput(value) {
+        this.emailInput = this.emailInput.val(value);
+    }
+
+    isValidEmail() {
+        // Regex with a email form
+        // eslint-disable-next-line no-useless-escape
+        // eslint-disable-next-line unicorn/better-regex
+        // eslint-disable-next-line no-useless-escape
+        const EmailRegex = /^([a-zA-Z0-9_.+-])+\@(([a-zA-Z0-9-])+\.)+([a-zA-Z0-9]{2,4})+$/;
+        if (!EmailRegex.test(this.emailInput.val())) {
+            throw new Error('not a valid e-mail address');
+        } else {
+            return true;
+        }
+    }
+
+    getRequired() {
+        return this.emailInput.prop('required');
+    }
+
+    setRequired(boolean) {
+        this.emailInput.prop('required', boolean);
+        this.required = boolean;
+    }
+
+    isRequired() {
+        if (!this.emailInput.prop('required')) {
+            this.required = false;
+            return this.required;
+        }
+
+        return this.required;
+    }
+
+    isEmailEnable() {
+        if (!this.emailInput.prop('disabled')) {
+            return true;
+        }
+
+        return false;
+    }
+}
```

Comment:
Add a blank line at the end of the file

Reply:
![image](https://user-images.githubusercontent.com/62728106/163231240-86c2d434-e918-4fc1-b7f1-36d48cd1e3f5.png)